File: app/src/main/java/pl/mosenko/sunnypodlaskie/persistence/entities/WeatherDataEntity.java
package pl.mosenko.sunnypodlaskie.persistence.entities;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import java.util.Date;

import pl.mosenko.sunnypodlaskie.persistence.dao.WeatherDataEntityDAO;

import static pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity.TABLE_NAME;

/**
 * Created by syk on 16.05.17.
 */

@DatabaseTable(tableName = TABLE_NAME, daoClass = WeatherDataEntityDAO.class)
public class WeatherDataEntity extends AbstractOrmLiteEntity {
    public static final String TABLE_NAME = "weather_data";
    public static final String CITY_COLUMN = "city";
    public static final String ICON_KEY_COLUMN = "icon_key";
    public static final String DESCRIPTION_COLUMN = "description";
    public static final String TEMPERATURE_COLUMN = "temperature";
    public static final String PRESSURE_COLUMN = "pressure";
    public static final String HUMIDITY_COLUMN = "humidity";
    public static final String WIND_SPEED_COLUMN = "wind_speed";
    public static final String WIND_DEGREE_COLUMN = "wind_degree";
    public static final String SUNRISE_COLUMN = "sunrise";
    public static final String SUNSET_COLUMN = "sunset";

    @DatabaseField(columnName = CITY_COLUMN, canBeNull = false)
    private String city;
    @DatabaseField(columnName = ICON_KEY_COLUMN, canBeNull = false)
    private String iconKey;
    @DatabaseField(columnName = DESCRIPTION_COLUMN)
    private String description;
    @DatabaseField(columnName = TEMPERATURE_COLUMN, canBeNull = false)
    private Double temperature;
    @DatabaseField(columnName = PRESSURE_COLUMN)
    private Double pressure;
    @DatabaseField(columnName = HUMIDITY_COLUMN)
    private Integer humidity;
    @DatabaseField(columnName = WIND_SPEED_COLUMN)
    private Double windSpeed;
    @DatabaseField(columnName = WIND_DEGREE_COLUMN)
    private Double windDegree;
    @DatabaseField(columnName = SUNRISE_COLUMN)
    private Date sunrise;
    @DatabaseField(columnName = SUNSET_COLUMN)
    private Date sunset;

    public WeatherDataEntity() {
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getIconKey() {
        return iconKey;
    }

    public void setIconKey(String iconKey) {
        this.iconKey = iconKey;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Double getTemperature() {
        return temperature;
    }

    public void setTemperature(Double temperature) {
        this.temperature = temperature;
    }

    public Double getPressure() {
        return pressure;
    }

    public void setPressure(Double pressure) {
        this.pressure = pressure;
    }

    public Integer getHumidity() {
        return humidity;
    }

    public void setHumidity(Integer humidity) {
        this.humidity = humidity;
    }

    public Double getWindSpeed() {
        return windSpeed;
    }

    public void setWindSpeed(Double windSpeed) {
        this.windSpeed = windSpeed;
    }

    public Double getWindDegree() {
        return windDegree;
    }

    public void setWindDegree(Double windDegree) {
        this.windDegree = windDegree;
    }

    public Date getSunrise() {
        return sunrise;
    }

    public void setSunrise(Date sunrise) {
        this.sunrise = sunrise;
    }

    public Date getSunset() {
        return sunset;
    }

    public void setSunset(Date sunset) {
        this.sunset = sunset;
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/persistence/entities/WeatherDataEntity.kt
package pl.mosenko.sunnypodlaskie.persistence.entities

import android.provider.BaseColumns
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import pl.mosenko.sunnypodlaskie.persistence.dao.WeatherDataEntityDAO
import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity
import java.util.*

/**
 * Created by syk on 16.05.17.
 */
@DatabaseTable(tableName = WeatherDataEntity.TABLE_NAME, daoClass = WeatherDataEntityDAO::class)
class WeatherDataEntity : AbstractOrmLiteEntity() {
    @DatabaseField(columnName = CITY_COLUMN, canBeNull = false)
    private var city: String? = null

    @DatabaseField(columnName = RECEIVING_TIME_COLUMN, canBeNull = false)
    private var receivingTime: Date? = null

    @DatabaseField(columnName = ICON_KEY_COLUMN, canBeNull = false)
    private var iconKey: String? = null

    @DatabaseField(columnName = WEATHER_CONDITION, canBeNull = false, foreignColumnName = BaseColumns._ID, foreign = true, foreignAutoRefresh = true)
    private var weatherCondition: WeatherConditionEntity? = null

    @DatabaseField(columnName = TEMPERATURE_COLUMN, canBeNull = false)
    private var temperature: Double? = null

    @DatabaseField(columnName = PRESSURE_COLUMN)
    private var pressure: Double? = null

    @DatabaseField(columnName = HUMIDITY_COLUMN)
    private var humidity: Int? = null

    @DatabaseField(columnName = WIND_SPEED_COLUMN)
    private var windSpeed: Double? = null

    @DatabaseField(columnName = WIND_DEGREE_COLUMN)
    private var windDegree: Double? = null

    @DatabaseField(columnName = SUNRISE_COLUMN)
    private var sunrise: Date? = null

    @DatabaseField(columnName = SUNSET_COLUMN)
    private var sunset: Date? = null
    fun getCity(): String? {
        return city
    }

    fun setCity(city: String?) {
        this.city = city
    }

    fun getReceivingTime(): Date? {
        return receivingTime
    }

    fun setReceivingTime(receivingTime: Date?) {
        this.receivingTime = receivingTime
    }

    fun getIconKey(): String? {
        return iconKey
    }

    fun setIconKey(iconKey: String?) {
        this.iconKey = iconKey
    }

    fun getWeatherCondition(): WeatherConditionEntity? {
        return weatherCondition
    }

    fun setWeatherCondition(weatherCondition: WeatherConditionEntity?) {
        this.weatherCondition = weatherCondition
    }

    fun getTemperature(): Double? {
        return temperature
    }

    fun setTemperature(temperature: Double?) {
        this.temperature = temperature
    }

    fun getPressure(): Double? {
        return pressure
    }

    fun setPressure(pressure: Double?) {
        this.pressure = pressure
    }

    fun getHumidity(): Int? {
        return humidity
    }

    fun setHumidity(humidity: Int?) {
        this.humidity = humidity
    }

    fun getWindSpeed(): Double? {
        return windSpeed
    }

    fun setWindSpeed(windSpeed: Double?) {
        this.windSpeed = windSpeed
    }

    fun getWindDegree(): Double? {
        return windDegree
    }

    fun setWindDegree(windDegree: Double?) {
        this.windDegree = windDegree
    }

    fun getSunrise(): Date? {
        return sunrise
    }

    fun setSunrise(sunrise: Date?) {
        this.sunrise = sunrise
    }

    fun getSunset(): Date? {
        return sunset
    }

    fun setSunset(sunset: Date?) {
        this.sunset = sunset
    }

    override fun toString(): String {
        return "WeatherDataEntity{" +
                "city='" + city + '\'' +
                ", receivingTime=" + receivingTime +
                ", iconKey='" + iconKey + '\'' +
                ", weatherCondition=" + weatherCondition +
                ", temperature=" + temperature +
                ", pressure=" + pressure +
                ", humidity=" + humidity +
                ", windSpeed=" + windSpeed +
                ", windDegree=" + windDegree +
                ", sunrise=" + sunrise +
                ", sunset=" + sunset +
                '}'
    }

    companion object {
        val TABLE_NAME: String? = "weather_data"
        val CITY_COLUMN: String? = "city"
        val RECEIVING_TIME_COLUMN: String? = "receiving_time"
        val ICON_KEY_COLUMN: String? = "icon_key"
        val WEATHER_CONDITION: String? = "weather_condition"
        val TEMPERATURE_COLUMN: String? = "temperature"
        val PRESSURE_COLUMN: String? = "pressure"
        val HUMIDITY_COLUMN: String? = "humidity"
        val WIND_SPEED_COLUMN: String? = "wind_speed"
        val WIND_DEGREE_COLUMN: String? = "wind_degree"
        val SUNRISE_COLUMN: String? = "sunrise"
        val SUNSET_COLUMN: String? = "sunset"
    }
}
